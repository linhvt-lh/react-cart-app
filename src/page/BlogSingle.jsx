import React, { useState } from "react";
import Button from "../component/Button/Button";

export default function BlogSingle(){
    const [postID, setPostID] = useState(null);
 
    if( !postID ){
        return  <>
        <h1>This is Blog Single Page</h1><h2 data-testid="loading">Loading...</h2>
        </>
    }
    return  <>
        <h1>This is Blog Single Page</h1>
        <Button></Button>
    </>

}