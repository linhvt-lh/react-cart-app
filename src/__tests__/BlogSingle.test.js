/**
 * @jest-environment jsdom
 */
import React from "react";
import BlogSingle from "../page/BlogSingle";
import { render, cleanup, screen } from "@testing-library/react";

afterEach(cleanup);

describe('Blog Single', () => {
    it ('Test snapshot testing', () => {
        const { asFragment } = render(<BlogSingle />);
        expect(asFragment(<BlogSingle />)).toMatchSnapshot();
    });

    it( 'Blog Single should render correctly', () => {
        let { getByTestId } = render(<BlogSingle />);
        expect( getByTestId('loading').textContent).toBe('Loading...');
    });
});