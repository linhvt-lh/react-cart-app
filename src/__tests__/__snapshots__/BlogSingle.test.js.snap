// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Single Test snapshot testing 1`] = `
<DocumentFragment>
  <h1>
    This is Blog Single Page
  </h1>
  <h2
    data-testid="loading"
  >
    Loading...
  </h2>
</DocumentFragment>
`;
